src-built-in/components/appMarket/src/stores/storeActions.js

```diff
@@ -0,0 +1,288 @@
+        let thisAppName = app.title ? app.title : app.name;
+        if (thisAppName === name && installed.includes(app.appId)) {
+            let index = installed.indexOf(app.appId);
+            installed.splice(index, 1);
+            break;
+        }
+    }
+
+    getStore().setValue({
+        field: 'installed',
+        value: installed
+    });
+}
+
+/**
+ * Gets the list of installed apps
+ */
+function getInstalledApps() {
+    return getStore().getValue({
+        field: 'installed'
+    });
+}
+
+/**
+ * Gets the list of filtered apps (when searching/filtering by tags)
+ */
+function getFilteredApps() {
+    return getStore().getValue({
+        field: 'filteredApps'
+    });
+}
+
+/**
+ * Clears the list of filtered apps
+ */
+function clearFilteredApps() {
+    getStore().setValue({
+        field: 'filteredApps',
+        value: []
+    });
+}
+
+/**
+ * Async call to appD to get the list of all tags
+ */
+function getTags() {
+    return appd.getTags()
+}
+
+/**
+ * Gets the list of active tags (these are tags that are actively filtering the content list)
+ */
+function getActiveTags() {
+    return getStore().getValue({
+        field: 'activeTags'
+    });
+}
+
+/**
+ * Adds an 'active tag' to the list of filtered tags
+ * @param {string} tag The tag name
+ */
+function addTag(tag) {
+    _addActiveTag(tag);
+}
+
+/**
+ * Removes an 'active tag' from the list of filtered tags
+ * @param {string} tag The tag name
+ */
+function removeTag(tag) {
+    _removeActiveTag(tag);
+}
+
+/**
+ * Removes all tags from the active tags list
+ */
+function clearTags() {
+    _clearActiveTags();
+}
+
+/**
+ * Calls appD to search the directory of apps based on search text and tag names
+ * @param {string} terms The search terms provided by the user
+ */
+function searchApps(terms) {
+    let activeTags = getStore().getValue({
+        field: 'activeTags'
+    });
+
+    appd.search({ text: terms, tags: activeTags }, (err, data) => {
+        if (err) console.log("Failed to search apps");
+        console.log('searching: ', data);
+        getStore().setValue({
+            field: 'filteredApps',
+            value: data
+        });
+    });
+}
+
+/**
+ * Retrieves a list of installed apps by id
+ */
+function getInstalledApps() {
+    return getStore().getValue({
+        field: 'installed'
+    });
+}
+
+/**
+ * Async function to call the launcher client to get a list of added apps
+ */
+function fetchInstalledApps() {
+    //NOTE: This is a WIP. This api may change so its commented out for now
+    // FSBL.Clients.LauncherClient.getComponentList((err, apps) => {
+    //     if (err) console.log("Error fetching list of added apps");
+    //     getStore().setValue({
+    //         field: 'installed',
+    //         value: apps
+    //     });
+    //     return apps;
+    // });
+    let apps = getStore().getValue({
+        field: 'apps'
+    });
+
+    let installed = [];
+    if (apps.length > 0) installed.push([apps[0].appId]);
```

Comment:
So this only ever has 0 or 1 installed apps?

Reply:
There is no way of knowing at the current moment how we will handle "installing" (or adding) apps. This is only a placeholder so one app is "installed" for UI testing purposes